php/class-customize-snapshot-manager.php

```diff
@@ -381,6 +389,235 @@ function suspend_kses_for_snapshot_revision_restore() {
 	}
 
 	/**
+	 * Filter for displaying the Snapshot menu location values.
+	 *
+	 * @param array $menu_locations Default menu locations.
+	 * @return array Modified menu locations.
+	 */
+	public function filter_theme_mod_nav_menu_locations( $menu_locations ) {
+		if ( false === $this->snapshot->is_preview() ) {
+			return $menu_locations;
+		}
+
+		$values = $this->snapshot->values();
+		$locations = get_registered_nav_menus();
+
+		foreach ( $locations as $location => $name ) {
+			if ( isset( $values[ 'nav_menu_locations[' . $location . ']' ] ) ) {
+				$menu_locations[ $location ] = $values[ 'nav_menu_locations[' . $location . ']' ];
+			}
+		}
+
+		return $menu_locations;
+	}
+
+	/**
+	 * Filter wp_get_nav_menus() to load Snapshot values.
+	 *
+	 * @see wp_get_nav_menus()
+	 *
+	 * @param array $menus Array of menus.
+	 * @return array Modified array of menus.
+	 */
+	public function filter_wp_get_nav_menus( $menus ) {
```

Comment:
I wonder why this is needed as opposed to what `WP_Customize_Nav_Menu_Setting::preview()` adds? https://github.com/xwp/wordpress-develop/blob/master/src/wp-includes/customize/class-wp-customize-nav-menu-setting.php#L204-L296